paimon-common/src/main/java/org/apache/paimon/table/SpecialFields.java

```diff
@@ -72,6 +72,8 @@ public class SpecialFields {
 
     public static final String KEY_FIELD_PREFIX = "_KEY_";
     public static final int KEY_FIELD_ID_START = SYSTEM_FIELD_ID_START;
+    // reserve 1000 for other system fields
+    public static final int KEY_FIELD_ID_END = Integer.MAX_VALUE - 1_000;
```

Comment:
So there can be at most $10^9$ keys and only $10^3$ system fields? What if I need to add a new group of system fields?